engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailAggregator.java

```diff
@@ -142,31 +177,107 @@ public Map<User, Map<String, Object>> getAggregated(UUID appId, EmailAggregation
             totalAggregatedElements += aggregations.size();
         } while (maxPageSize == aggregations.size());
         Log.infof("%d elements were aggregated for key %s", totalAggregatedElements, aggregationKey);
+    }
 
-        return aggregated.entrySet().stream()
-                .filter(entry -> !entry.getValue().isEmpty())
-                .collect(toMap(
-                        Entry::getKey,
-                        entry -> entry.getValue().getContext()
-                ));
+    private void aggregationBasedOnEvent(EventAggregationCriteria eventAggregationCriteria, LocalDateTime start, LocalDateTime end, Map<String, Set<String>> subscribersByEventType, Map<String, Set<String>> unsubscribersByEventType, Map<User, AbstractEmailPayloadAggregator> aggregated) {
```

Comment:
In the old aggregation way, we're transforming the event before aggregating them: https://github.com/RedHatInsights/notifications-backend/blob/4459d29fc7ff7ef6e8c42097d4d7a3912e7a26b0/engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailAggregationProcessor.java#L161

Do we need that transformation with the new aggregation way?

Reply:
I checked, we don't.

Reply:
Thanks for checking!